### ISSUE TITLE
[vmagent] Panics and strange behavior when upgrade to 1.93.1+

### ISSUE THREAD

**Issue by Sinketsu** [VictoriaMetrics/VictoriaMetrics#4982]
### Describe the bug

Hello! 
We wanted to upgrade our Victoria cluster, but encountered a strange `vmagent` operation after upgrading to 1.93.1 or higher. We have not changed the configs or the load on it.
He pushes very large values into the storage to existing timeseries and sometimes panics.

We have set `-remoteWrite.label="prom_cluster=testing"` flag and I think the errors are related to patches for another issue https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4247.
In panic we have log like:
```
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/remotewrite.go:450 +0xe8   
created by github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.pushBlockToRemoteStorages in goroutine 112   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/remotewrite.go:452 +0x55   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.pushBlockToRemoteStorages.func2(0xc027c38900?)   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/remotewrite.go:694 +0x5df   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.(*remoteWriteCtx).Push(0xc000180de0, {0xc0333ec000?, 0x0?, 0xc005c7c420?})   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/remotewrite.go:730 +0x109   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.(*remoteWriteCtx).pushInternal(0xc000180de0?, {0xc0333ec000?, 0x1e5?, 0x355?})   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/pendingseries.go:62 +0x65   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.(*pendingSeries).Push(0xc0000003c0, {0xc0333ec000?, 0x1e5?, 0x0?})   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/pendingseries.go:164 +0x186   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.(*writeRequest).push(0xc0000003c8, {0xc0333ec000, 0x1e5, 0xc034449da0?})   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/pendingseries.go:185 +0x67b   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.(*writeRequest).copyTimeSeries(0xc0000003c8, 0xc023d02000, 0xc0333eeb80)   
goroutine 325767 [running]:   
panic: runtime error: slice bounds out of range [-6:]
```

or 

```
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/remotewrite.go:450 +0xe8   
created by github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.pushBlockToRemoteStorages in goroutine 122   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/remotewrite.go:452 +0x55   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.pushBlockToRemoteStorages.func2(0xc03f812900?)   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/remotewrite.go:694 +0x5df   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.(*remoteWriteCtx).Push(0xc000280cc0, {0xc047cf9000?, 0x0?, 0x0?})   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/remotewrite.go:730 +0x109   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.(*remoteWriteCtx).pushInternal(0xc000280cc0?, {0xc047cf9000?, 0xde?, 0x100?})   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/pendingseries.go:62 +0x65   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.(*pendingSeries).Push(0xc0000000c0, {0xc047cf9000?, 0xde?, 0xe7f160?})   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/pendingseries.go:164 +0x186   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.(*writeRequest).push(0xc0000000c8, {0xc047cf9000, 0xde, 0xc040ddfd88?})   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite/pendingseries.go:184 +0x2b0   
github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/remotewrite.(*writeRequest).copyTimeSeries(0xc0000000c8, 0xc01bb465b0, 0xc047cf96f0)   
goroutine 927395 [running]:   
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1 pc=0x46ed3c]  
panic: runtime error: invalid memory address or nil pointer dereference
```

Thank you!

### To Reproduce

I'm sorry, but I can't provide an example to reproduce. I can't reproduce this locally, only on our production load.

### Version

```
$ vmagent --version
vmagent-20230823-122625-tags-v1.93.1-0-gc8c20b7f7a
```

### Used command-line flags

-remoteWrite.label="prom_cluster=testing"

**Comment by f41gh7** [comment]
I'll take a look. Maybe related issue https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4961

**Comment by f41gh7** [comment]
Bug was introduced at v1.93.3 version. Will be fixed at the next release.

 Workaround for now, use v1.93.1 version.

**Comment by valyala** [comment]
This issue seems to be related to https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4972

**Comment by valyala** [comment]
Closing this issue as duplicate for https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4972 . See more details at https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4972#issuecomment-1712412380

**Comment by valyala** [comment]
FYI, the issue should be fixed in [v1.93.4](https://docs.victoriametrics.com/CHANGELOG.html).
